File: Backend/SOS-Clean-House-back/src/main/java/com/SOS_Clean_House_back/model/EnderecoServico.java
package com.SOS_Clean_House_back.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor @NoArgsConstructor @Data @Builder


public class EnderecoServico {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String rua;
    private Integer numero;
    private String bairro;
    private String cidade;
    private Integer cep;
    private String observacao;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @OneToOne
    @JoinColumn(name = "agendamento_id")
    private Agendamento enderecoAgendamento;
}

File: Frontend/SOS-Clean-House-front/src/app/componentes/interface/EnderecoServico.ts
import { Agendamento } from "./Agendamento";
import { Cliente } from "./Cliente";
import { Servico } from "./Servico";

export interface EnderecoServico {
    id: number;
    rua: string;
    numero: number;
    bairro: string;
    cidade: string;
    cep: number;
    observacao: string;
    cliente: Cliente;
    enderecoAgendamento: Agendamento;
    servicoReference: Servico;
  }